src/evm/protocol/vm/state_builder.rs

```diff
@@ -221,26 +230,53 @@ where
         } else {
             self.get_default_capabilities()?
         };
-        Ok(EVMPoolState::new(
-            self.id,
-            self.tokens,
-            self.block,
-            self.balances,
-            self.balance_owner,
-            HashMap::new(),
-            capabilities,
-            HashMap::new(),
-            self.involved_contracts
-                .unwrap_or_default(),
-            self.token_storage_slots
-                .unwrap_or_default(),
-            self.manual_updates.unwrap_or(false),
-            self.adapter_contract.ok_or_else(|| {
-                SimulationError::FatalError(
-                    "Failed to get build engine: Adapter contract not initialized".to_string(),
-                )
-            })?,
-        ))
+
+        let adapter_contract = self.adapter_contract.ok_or_else(|| {
+            SimulationError::FatalError(
+                "Failed to get build engine: Adapter contract not initialized".to_string(),
+            )
+        })?;
+
+        if !self.contract_balances.is_empty() {
```

Comment:
Not sure I like this, especially if we keep supporting both we may have other options here?